modules/common/secret/secret.go

```diff
@@ -165,7 +165,13 @@ func createOrUpdateSecret(
 		// Note: this can overwrite data rendered from GetTemplateData() if key is same
 		if len(st.CustomData) > 0 {
 			for k, v := range st.CustomData {
-				dataString[k] = v
+				vExpanded, err := util.ExecuteTemplateData(v, st.ConfigOptions)
```

Comment:
It should be possible to consolidate this code and the code for the config map into a single function, but I don't think that's a big deal.